arelle/Locale.py

```diff
@@ -98,23 +86,145 @@ def getUserLocale(localeCode: str = '') -> tuple[LocaleDict, str | None]:
 
 
 def getLanguageCode() -> str:
-    if sys.platform == "darwin": # MacOS doesn't provide correct language codes
-        localeQueryResult = subprocess.getstatusoutput("defaults read -g AppleLocale")  # MacOS only
-        if localeQueryResult[0] == 0 and localeQueryResult[1]: # successful
-            return localeQueryResult[1][:5].replace("_","-")
-    import locale
-    languageCode, encoding = locale.getdefaultlocale()
-    # language code and encoding may be None if their values cannot be determined.
-    if isinstance(languageCode, str):
-        return languageCode.replace("_","-")
+    if systemLocale := getLocale():
+        languageCode = langCodeXBRLFormat(systemLocale)
+        if re.match(BCP47_LANGUAGE_CODE, languageCode):
+            return languageCode
     from arelle.XbrlConst import defaultLocale
-    return defaultLocale # XBRL international default locale
+    return defaultLocale  # XBRL international default locale
+
 
 def getLanguageCodes(lang: str | None = None) -> list[str]:
+    """
+    Returns a list of language code formats that can be used with gettext to look up translation files.
+    [en_US, en-US, en]
+    [fr]
+    """
     if lang is None:
         lang = getLanguageCode()
-    # allow searching on the lang with country part, either python or standard form, or just language
-    return [lang, lang.replace("-","_"), lang.partition("-")[0]]
+    if LANGUAGE_CODE_LOCALE_SEPARATOR in lang:
+        return [lang, langCodeXBRLFormat(lang), lang.partition(LANGUAGE_CODE_LOCALE_SEPARATOR)[0]]
```

Comment:
I'm a bit confused about the terminology here.  I think the relevant string formats are "POSIX locale" and BCP 47 "language tag".  Is the latter what you're calling `LANGUAGE_CODE_XML_SEPARATOR` / `langCodeXBRLFormat`?  If so, perhaps `LANGUAGE_TAG_SEPARATOR`, `langTagFormat`, etc. would be clearer names.

The `lang` param sounds like it's language tag, but the code checks for a locale string separator, and the caller is passing a var named `locale`, which seems to come from a user-specified string.  Maybe it's really just a fuzzy config string that might be lang or locale and `configLang` would be clearer, or perhaps just a small doc comment that the string is user-specified, so we're making a best effort.

Reply:
I updated a bunch of the variable names and included comments. It should be much more clear now.